src/lightning/data/streaming/functions.py

```diff
@@ -56,8 +61,11 @@ def _get_input_dir(inputs: Sequence[Any]) -> Optional[str]:
 
     absolute_path = str(Path(list(indexed_paths.values())[0]).resolve())
 
+    if "/.project" in absolute_path:
```

Comment:
It looks like _get_input_dir is not tested. There is also no test_functions.py. Maybe we could add one to cover these cases now?

Reply:
I added one.